### PR TITLE
Add a simple Jenkinsfile for building on ci.jenkins.io

### PR DESCRIPTION
**Pull request by rtyler**
This should be able to be replaced at some point in the (hopefully) near future
by some steps in our shared library
(https://github.com/jenkins-infra/pipeline-library) but that's not quite ready
yet

### CODE REVIEW

**Comment by rtyler** [comment]
@oleg-nenashev I believe the failure in jenkins.ci.c.c is unrelated, but let me know if that's not the case and I can dig further

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        }
+
+        stage('Build') {
+            withEnv([
+                "JAVA_HOME=${tool 'jdk8'}",
+                "PATH+MVN=${tool 'mvn'}/bin",
+                'PATH+JDK=$JAVA_HOME/bin',
+            ]) {
+                timeout(30) {
+                    sh 'mvn clean install -Dmaven.test.failure.ignore=true'
```
Comment (oleg-nenashev) [inline Jenkinsfile]: Why do you ignore the test failure here?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            checkout scm
+        }
+
+        stage('Build') {
+            withEnv([
+                "JAVA_HOME=${tool 'jdk8'}",
+                "PATH+MVN=${tool 'mvn'}/bin",
+                'PATH+JDK=$JAVA_HOME/bin',
+            ]) {
+                timeout(30) {
```
Comment (oleg-nenashev) [inline Jenkinsfile]: I would set a 60-min timeout just in case the build queue is packed.

File: Jenkinsfile (hunk 1 above)
Comment (oleg-nenashev) [inline Jenkinsfile]: Ah, just because we do not want the build to fail...

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                    sh 'mvn clean install -Dmaven.test.failure.ignore=true'
+                }
+            }
+        }
+
+        stage('Archive') {
+            /* Archive the test results */
+            junit '**/target/surefire-reports/TEST-*.xml'
```
Comment (oleg-nenashev) [inline Jenkinsfile]: Should be the last invocation, which fails the build in the case of test errors

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env groovy
```
Comment (oleg-nenashev) [inline Jenkinsfile]: Not required anymore

**Comment by oleg-nenashev** [comment]
Build failure is unrelated. CC @stephenc, who has written this test

File: Jenkinsfile (hunk 2 above)
Comment (rtyler) [inline Jenkinsfile]: 30 minutes is the existing timeout value in the freestyle job

File: Jenkinsfile (hunk 4 above)
Comment (rtyler) [inline Jenkinsfile]: I add this out of habit to ensure the editor knows how to syntax highlight the document.

File: Jenkinsfile (hunk 2 above)
Comment (oleg-nenashev) [inline Jenkinsfile]: Which times out sometimes

File: Jenkinsfile (hunk 3 above)
Comment (rtyler) [inline Jenkinsfile]: This does not fail the Pipeline if there are test failures, it just marks the build as unstable. The `archiveArtifacts` step will still execute (as far as I am aware)

File: Jenkinsfile (hunk 2 above)
Comment (rtyler) [inline Jenkinsfile]: Is that not the correct behavior? Taking more than 30 minutes sounds like a problem with tests taking a long time. If this is going to spin for 60 minutes and still fail then that's costing us double the amount of compute $$$ :frowning:

File: Jenkinsfile (hunk 4 above)
Comment (oleg-nenashev) [inline Jenkinsfile]: OK

File: Jenkinsfile (hunk 2 above)
Comment (oleg-nenashev) [inline Jenkinsfile]: OK, let's keep it as is. At least it does not include the queue wait time

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+
+node('java') {
+    timestamps {
+        stage('Checkout') {
+            checkout scm
+        }
+
+        stage('Build') {
+            withEnv([
```
Comment (stephenc) [inline Jenkinsfile]: isn't `withMaven` supposed to set all this up for you (with the advantage of knowing about the settings) (not that I like the `withMaven` step... but still)

**Comment by stephenc** [comment]
LGTM :bee:

File: Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env groovy
+
+/* Only keep the 10 most recent builds. */
+properties([[$class: 'BuildDiscarderProperty',
+                strategy: [$class: 'LogRotator', numToKeepStr: '10']]])
+
+
+node('java') {
```
Comment (jtnord) [inline Jenkinsfile]: windows... windows windows...

File: Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+                }
+
+                stage('Build') {
+                    withEnv([
+                        "JAVA_HOME=${tool 'jdk8'}",
+                        "PATH+MVN=${tool 'mvn'}/bin",
+                        'PATH+JDK=$JAVA_HOME/bin',
+                    ]) {
+                        timeout(30) {
+                            sh 'mvn clean install -Dmaven.test.failure.ignore=true'
```
Comment (jtnord) [inline Jenkinsfile]: sh - on windows?

File: Jenkinsfile [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+                    checkout scm
+                }
+
+                stage('Build') {
+                    withEnv([
+                        "JAVA_HOME=${tool 'jdk8'}",
+                        "PATH+MVN=${tool 'mvn'}/bin",
+                        'PATH+JDK=$JAVA_HOME/bin',
+                    ]) {
+                        timeout(30) {
```
Comment (jtnord) [inline Jenkinsfile]: timeout is too short to be usefull when tests fail (due to retrying the tests multiple times).

Takes 34ish minutes on my machine with gen6 i7 and SSD :-/

File: Jenkinsfile [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+                            sh 'mvn clean install -Dmaven.test.failure.ignore=true'
+                        }
+                    }
+                }
+
+                stage('Archive') {
+                    /* Archive the test results */
+                    junit '**/target/surefire-reports/TEST-*.xml'
```
Comment (jtnord) [inline Jenkinsfile]: not that I care at this point (I wanta `Jenkinsfile` that builds on windows slaves) but what about findbugs?

File: Jenkinsfile (hunk 7 above)
Comment (rtyler) [inline Jenkinsfile]: Oh right, I forgot. It's still maddening that `bat` and `sh` are different steps

File: Jenkinsfile (hunk 8 above)
Comment (rtyler) [inline Jenkinsfile]: a little late to [this discussion](https://github.com/jenkinsci/remoting/pull/114?#discussion_r81625991)

File: Jenkinsfile (hunk 9 above)
Comment (rtyler) [inline Jenkinsfile]: not in this PR

**Comment by jtnord** [comment]
@rtyler @oleg-nenashev ship-it please!!

File: Jenkinsfile (hunk 7 above)
Comment (jtnord) [inline Jenkinsfile]: Yes would be nice to have something akin to xshell support for pipeline (running a durable task...)

**Comment by rtyler** [comment]
[SHAZAM](https://ci.jenkins.io/job/Core/job/remoting/)

**Pull request merged**